src/reflow/mod.rs

```diff
@@ -0,0 +1,542 @@
+//! Re-wrap documentation comments to a desired line width.
+//!
+//! Will handle hyphenation eventually if desired.
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use crate::checker::Checker;
+use crate::{CheckableChunk, ContentOrigin, Documentation};
+use crate::{Detector, Suggestion, SuggestionSet};
+
+/// Parameters for wrapping doc comments
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ReflowConfig {
+    /// Hard limit for absolute length of lines
+    max_line_length: usize,
+}
+
+impl Default for ReflowConfig {
+    fn default() -> Self {
+        ReflowConfig {
+            max_line_length: 80,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Reflow;
+
+impl Reflow {
+    /// Parse CommonMark files and reflow only neccessary portions of it
+    fn reflow_commonmark_chunks<'a, 's>(
+        origin: &ContentOrigin,
+        chunks: &'a Vec<CheckableChunk>,
+        config: &ReflowConfig,
+    ) -> Result<Vec<Suggestion<'s>>>
+    where
+        'a: 's,
+    {
+        let wrapper = textwrap::Wrapper::new(config.max_line_length)
+            .subsequent_indent("")
+            .initial_indent("");
+        unimplemented!();
+    }
+
+    /// Reflow chunks from rust source code
+    fn reflow_chunks<'a, 's>(
+        origin: &ContentOrigin,
+        chunks: &'a Vec<CheckableChunk>,
+        config: &ReflowConfig,
+    ) -> Result<Vec<Suggestion<'s>>>
+    where
+        'a: 's,
+    {
+        let wrapper = textwrap::Wrapper::new(config.max_line_length)
+            .subsequent_indent(" ")
+            .initial_indent(" ");
+        let suggestions = chunks
+            .iter()
+            .filter_map(|chunk| {
+                let mut new_lines = chunk
```

Comment:
I think this is where you would need to plug the markdown parser (see review comment).

Keep in mind that we need to preserver spaces, so I think we should just take markdown elements (including their style element extents) and just reflow those blocks. That makes things a lot simpler, and the parser should only have two logic blocks - elements with scope and elements without. You don't really care about which element you have, just if it contains others which one might want to reflow, or if it's an atomic element (i.e. a link).

Hope that makes sense somewhat :}

Reply:
I have no idea if the above approach is compatible with the crate that's used right now..

Reply:
I don't think this works, we need a hybrid of `textwrap` being able to derive placement from an initial offset + span or offset + range, deeply integrated with the commonmark parser.